### PR TITLE
chore: Move `react-native-async-storage` to `peerDependencies`

### PR DESCRIPTION
**Pull request by alphatrl**
## Purpose

The aim is to reduce the overall module size by moving `react-native-async-storage` to a `peerDependency`. Package users now need to install `react-native-async-storage` as well.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,18 +18,17 @@
...
-    "@react-native-async-storage/async-storage": "^1.13.3",
-    "@taskade/eslint-plugin": "^0.2.0",
     "emoji-datasource": "^6.0.0"
   },
-  "repository": {
-    "type": "git",
-    "url": "git+https://github.com/taskade/react-native-emoji-selector.git"
+  "peerDependencies": {
+    "@react-native-async-storage/async-storage": "^1.13.0",
+    "react-native": "^0.0.0-0 || 0.60 - 0.71 || 1000.0.0"
```
Comment (duncanleo) [inline package.json]: nit: would `>0.64.0` work here?

the semver range can be tested with the [npm semver calculator](https://semver.npmjs.com/)

File: package.json [hunk 2]
```diff
@@ -14,18 +18,17 @@
...
   },
   "devDependencies": {
     "@babel/core": "^7.12.10",
     "@babel/eslint-parser": "^7.12.1",
     "@babel/runtime": "^7.12.5",
+    "@taskade/eslint-plugin": "^0.2.0",
```
Comment (duncanleo) [inline package.json]: nit: could the ESLint changes be separated out to a different PR?

Comment (alphatrl) [reply package.json]: will add eslint changes and react-native bump to another PR

**Review by duncanleo — APPROVED** [review]
LGTM!
